src/traversals/bfs.jl

```diff
@@ -263,14 +263,32 @@ end
 """Will return `true` if graph `g` is
 [bipartite](https://en.wikipedia.org/wiki/Bipartite_graph).
 """
-function is_bipartite(g::SimpleGraph, s::Int)
+is_bipartite(g::SimpleGraph, s::Int) = bipartite_internal(g, s).is_bipartite
+
+function is_bipartite(g::SimpleGraph)
+    cc = filter(x->length(x)>2, connected_components(g))
+    return all(x->is_bipartite(g,x[1]), cc)
+end
+
+function bipartite_internal(g::SimpleGraph, s::Int)
     nvg = nv(g)
     visitor = BipartiteVisitor(nvg)
     traverse_graph(g, BreadthFirst(), s, visitor)
-    return visitor.is_bipartite
+    return visitor
 end
 
-function is_bipartite(g::SimpleGraph)
-    cc = filter(x->length(x)>2, connected_components(g))
-    return all(x->is_bipartite(g,x[1]), cc)
+"""
+If the graph is bipartite returns a vector `c`  of size `nv(g)` containing
+the assignment of each vertex to one of the two sets (`c[i] == 1` or `c[i]==2`).
+If `g` is not bipartite returns an empty vector.
+"""
+function bipartite_map(g::SimpleGraph)
+    cc = connected_components(g)
```

Comment:
Let's see if we can make this consistent with parity in https://github.com/JuliaGraphs/LightGraphs.jl/blob/master/src/traversals/maxadjvisit.jl#L176-L189 where we return a vector of booleans.